File: functions/financial_calc_update.py
# Malu Estevam, Financial Calculator Updated Python

def info(income, amount, type):
    type_percent = (amount/income)*100
    print(f"You spend ${amount:.2f} on {type} and that is {type_percent:.2f}% of your income")

def user(type):
    return float(input(f"What is the cost of your {type}\n"))
    
# write a print statement telling the user what the program is (budget calculator)
print("Welcome to your financial calculator. Follow the instructions below")
# Ask for monthly income (user input) 
income = user("income")
# Ask for rent amount (user input) 
rent = user("rent")
# Ask for utilities amount (user input) 
utilities = user("utilities")
# Ask for groceries amount (user input) 
groceries = user("groceries")
# Ask for transportation amount (user input)
transportation = user("transportation")
# Calculate savings as 10% of income (variable)
savings = float(income/10)
# Calculate spending money income - (rent+utilities+groceries+transportation+savings)*100 (variable)
spend = (income - (rent+utilities+groceries+transportation+savings))
spend_percent = spend/income*100


info(income, rent, "rent")
info(income, utilities, "utilities")
info(income, groceries, "groceries")
info(income, transportation, "transportation")
print(f"You should save ${savings:.2f} and that is 10% of your income")
print(f"You have ${spend:.2f} left to spend and that is {spend_percent:.2f}% of your income.")

File: functions/financial_calc_update.c
//Malu Estevam, Financial Calculator C
#include <stdio.h>
#include <math.h>

float income, rent, utilities, groceries, transportation, spend, savings;

float user(char type[]){
    float amount; 
    printf("What is your %s amount?\n", type); 
    scanf("%f", &amount);
    return amount;
}

void info(float income, float amount){
    float percent = (amount/income)*100;
    printf("You spend $%.2f on rent which is %.2f%% your income\n", amount, percent);

}
int main(void){
    printf("Welcome to your very own financial calculator. Follow the instructions below\n");

    income = user("income");

    rent = user("rent");

    utilities = user("utilities");

    groceries = user("groceries");

    transportation = user("transportation");

    float savings = income/10;
    float spend = income - (rent+utilities+groceries+transportation+savings);
    float spend_percent = spend/income*100;


    info(income, rent);
    info(income, utilities);
    info(income, groceries);
    info(income, transportation);
    printf("You have $%.2f for savings which is 10%% of your income\n", savings);
    printf("You have $%.2f left for spending which is %.2f%% of your income\n", spend, spend_percent);

    return 0;
}